File: c/eu005.c
// project euler (projecteuler.net) problem 5:
// solution by Kevin Retzke (retzkek@gmail.com), November 2009

#include <stdio.h>
#include <stdbool.h>

#include "pe.h"

// Calculates the smallest number that is evenly divisible by all 
// integers from 1 to max (inclusive).
int smallestDivisibleBy(int max)
{
	int i;
	bool candidate, found;
	i = 0;
	found = false;
	while (!found) {
		i += max;
		candidate = true;
		for (int j = max; j > max/2-1; j--) {
			if (i%j != 0) {
				candidate = false;
				break;
			}
		}
		found = candidate;
	}
	return i;
}

int main()
{
	pe_test_eq(smallestDivisibleBy(10), 2520, "%d");
	printf("%d\n", smallestDivisibleBy(20));
}

File: julia/eu005.jl
#!/usr/bin/env julia

# divisible_by_all returns the smallest integer that is evenly divisible
# by all integers in ns.
function divisible_by_all(ns::Array{Int})
    i = 0
    step = maximum(ns)
    found = false
    while !found
        i += step
        candidate = true
        for n in ns
            if i%n != 0
                candidate = false
                break
            end
        end
        found = candidate
    end
    return i
end

using Test
@test divisible_by_all(collect(1:10)) == 2520

# problem
println(divisible_by_all(collect(1:20)))
